syntax = "proto3";
package bringyour;

option go_package = "bringyour.com/protocol";

// Converts a struct of type map[tid]map[tid]uint64

message CoocInner {
    string sid = 1;  // inner tid
    uint64 overlap = 2; // inner map value
}

message CoocOuter {
    string sid = 1;  // outer tid
    repeated CoocInner cooc_inner = 2; // list of inner maps
}

message CooccurrenceData {
    repeated CoocOuter cooc_outer = 1; // list of outer maps
}

message TimesData {
    repeated Times times = 1; // list of outer maps
}

message Times {
    string sid = 1;  // tid
    repeated uint64 time = 2; // list of times
}

message ClustersData {
    repeated Cluster clusters = 1; // list of clusters
}

message Cluster {
    string id = 1; // cluster id
    repeated string sids = 2; // list of tids
}